src/format/strftime.rs

```diff
@@ -466,28 +466,45 @@ impl<'a> Iterator for StrftimeItems<'a> {
                 }
             }
 
-            // the next item is space
+            // whitespace
             Some(c) if c.is_whitespace() => {
-                // `%` is not a whitespace, so `c != '%'` is redundant
-                let nextspec = self
-                    .remainder
-                    .find(|c: char| !c.is_whitespace())
-                    .unwrap_or(self.remainder.len());
-                assert!(nextspec > 0);
-                let item = sp!(&self.remainder[..nextspec]);
-                self.remainder = &self.remainder[nextspec..];
+                let ws = self.remainder;
```

Comment:
Could this be a different commit? Please get rid of the `_` suffix.

Reply:
> Could this be a different commit?

I re-ran the old code and I could not elicit a test case that justified the new code. I'm pretty sure I added the code early-on in this PR endeavor to fix a multi-byte unicode character parsing problem. But later, I fixed something else such that this change was unnecessary.  So I reverted my changes.

I left the added test cases to have some assurance about my claim; multi-byte characters are correctly processed.